lib/listen/record.rb

```diff
@@ -64,7 +64,10 @@ def _sub_tree(rel_path)
         if path == rel_path
           result.merge!(meta)
         else
-          sub_path         = path.sub(%r{\A#{rel_path}/?}, '')
```

Comment:
Also it was a bug that the `rel_path` wasn't escaped before interpolating it into a regexp. All sorts of weird things could happen if `rel_path` had characters in it that are special in regexps, like ^ $ [ \ ? | . * etc.

Reply:
Yeah, I generally find regexes super hard to grok, so I'm a fan of removing it in favour of more explicit code. That's a matter of personal preference though - I can go back to a regex-based solution if you prefer.

Reply:
I agree with you! It is best to avoid regexps in code. My experience with them is that the likelihood of being 100% correct drops quadratically with every punctuation character used. :)